graylog2-web-interface/src/components/common/ExternalLink.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+
+import Icon from './Icon';
+
+type Props = {
+  href?: string,
+  children: React.ReactNode,
+  target?: string,
+  iconClass?: string,
+  className?: string,
+};
+
+/**
+ * Component that renders a link to an external resource.
+ */
+const ExternalLink = ({ children, className, href, iconClass, target }: Props) => {
+  const content = (
+    <span>
+      {children}
+      &nbsp;
+      <Icon name={iconClass} />
+    </span>
+  );
+
+  // This makes the component usable as child element of a component that already renders a link (e.g. MenuItem)
+  if (_.trim(href) === '') {
+    return content;
+  }
+
+  return (
+    <a href={href} target={target} className={className}>
```

Comment:
External links should include `noopener noreferrer` incase they are using an older browser. Reduces potential of leaking sensitive data.


```suggestion
    <a href={href} target={target} className={className} ref="noopener noreferrer">
```

Reply:
:+1: - IIRC it's `rel`, not `ref`. We have an ESLint rule in place enforcing it if `target="_blank"` (but not in this case).

Reply:
Oops - muscle-memory typo! You're correct on the `rel`.